static/js/analytics.js

```diff
@@ -215,11 +282,11 @@ const renderChartsFromSingleUserJson = (json) => {
   renderTaggedFriendsGraphFromJson(json);
 };
 
-const getAndRenderAnalytics = async () => {
-
-  const requestUrl = `${baseUrl}/${API_PREFIX}${pathName}`;
+const getAndRenderAnalytics = async (startDate, endDate) => {
```

Comment:
Make startDate and endDate an optional  param, this simplify the logic also below(eg line 335 / 325)

Reply:
I think this is related with the misunderstanding about the "|" operator

Reply:
Nope, different thing. I'd make it so you are not required to pass both param when calling it.